http/handler.go

```diff
@@ -0,0 +1,138 @@
+			tx.Delete("/{id}", tx_handler.Discard)
+		})
+		api.Route("/backup", func(backup chi.Router) {
+			backup.Post("/export", store_handler.BasicExport)
+			backup.Post("/import", store_handler.BasicImport)
+		})
+		api.Route("/schema", func(schema chi.Router) {
+			schema.Post("/", store_handler.AddSchema)
+			schema.Patch("/", store_handler.PatchSchema)
+			schema.Post("/default", store_handler.SetDefaultSchemaVersion)
+		})
+		api.Route("/collections", func(collections chi.Router) {
+			collections.Get("/", store_handler.GetCollection)
+			// with collection middleware
+			collections_tx := collections.With(CollectionMiddleware)
+			collections_tx.Get("/{name}", collection_handler.GetAllDocKeys)
+			collections_tx.Post("/{name}", collection_handler.Create)
+			collections_tx.Patch("/{name}", collection_handler.UpdateWith)
+			collections_tx.Delete("/{name}", collection_handler.DeleteWith)
+			collections_tx.Post("/{name}/indexes", collection_handler.CreateIndex)
+			collections_tx.Get("/{name}/indexes", collection_handler.GetIndexes)
+			collections_tx.Delete("/{name}/indexes/{index}", collection_handler.DropIndex)
+			collections_tx.Get("/{name}/{key}", collection_handler.Get)
+			collections_tx.Patch("/{name}/{key}", collection_handler.Update)
+			collections_tx.Delete("/{name}/{key}", collection_handler.Delete)
+		})
+		api.Route("/lens", func(lens chi.Router) {
+			lens.Use(LensMiddleware)
+			lens.Get("/", lens_handler.Config)
+			lens.Post("/", lens_handler.SetMigration)
+			lens.Post("/reload", lens_handler.ReloadLenses)
+			lens.Get("/{version}", lens_handler.HasMigration)
+			lens.Post("/{version}/up", lens_handler.MigrateUp)
+			lens.Post("/{version}/down", lens_handler.MigrateDown)
+		})
+		api.Route("/graphql", func(graphQL chi.Router) {
+			graphQL.Get("/", store_handler.ExecRequest)
+			graphQL.Post("/", store_handler.ExecRequest)
+		})
+		api.Route("/ccip", func(ccip chi.Router) {
+			ccip.Get("/{sender}/{data}", ccip_handler.ExecCCIP)
+			ccip.Post("/", ccip_handler.ExecCCIP)
+		})
+		api.Route("/p2p", func(p2p chi.Router) {
+			p2p.Get("/info", store_handler.PeerInfo)
+			p2p.Route("/replicators", func(p2p_replicators chi.Router) {
+				p2p_replicators.Get("/", store_handler.GetAllReplicators)
+				p2p_replicators.Post("/", store_handler.SetReplicator)
+				p2p_replicators.Delete("/", store_handler.DeleteReplicator)
+			})
+			p2p.Route("/collections", func(p2p_collections chi.Router) {
+				p2p_collections.Get("/", store_handler.GetAllP2PCollections)
+				p2p_collections.Post("/{id}", store_handler.AddP2PCollection)
+				p2p_collections.Delete("/{id}", store_handler.RemoveP2PCollection)
+			})
+		})
+		api.Route("/debug", func(debug chi.Router) {
+			debug.Get("/dump", store_handler.PrintDump)
+		})
+	})
+
+	router.Handle("/*", playgroundHandler)
+
+	return &Handler{
+		db:     db,
+		router: router,
+		txs:    txs,
+	}
+}
+
+func (h *Handler) Transaction(id uint64) (datastore.Txn, error) {
```

Comment:
This was added so the CLI tests have access to the transactions. This should be cleaned up when we switch to an internal transaction pool.